### PR TITLE
Fixes to how output items are displayed

### PR DESCRIPTION
**Pull request by DonJayamanne**

### CODE REVIEW

File: src/client/datascience/jupyter/kernels/cellExecution.ts [hunk 1]
```diff
@@ -296,8 +296,8 @@ export class CellExecution {
         await chainWithPendingUpdates(this.editor.document, (edit) => {
             traceCellMessage(this.cell, 'Update cell statue as idle and message as empty');
             const metadata = this.cell.metadata.with({
-                runStartTime: undefined,
-                lastRunDuration: undefined,
+                runStartTime: null,
```
Comment (DonJayamanne) [inline src/client/datascience/jupyter/kernels/cellExecution.ts:299]: Changes in API.

File: src/client/datascience/jupyter/kernels/cellExecution.ts [hunk 2]
```diff
@@ -632,25 +632,27 @@ export class CellExecution {
...
-            );
-            // Get the jupyter output from the vs code output (so we can concatenate the text ourselves).
-            const outputs = existingItemToBeReplaced ? createIOutputFromCellOutputs([existingItemToBeReplaced]) : [];
-            if (existingItemToBeReplaced && outputs.length === 1 && nbformat.isStream(outputs[0])) {
-                let existingOutputText: string = concatMultilineString((outputs[0] as nbformat.IStream).text);
+            // Ensure we append to previous output, only if the streams as the same &
+            // If the last output is the desired stream type.
+            const lastOutput =
+                exitingCellOutputs.length > 0 ? exitingCellOutputs[exitingCellOutputs.length - 1] : undefined;
+            const existingOutputToAppendTo =
```
Comment (DonJayamanne) [inline src/client/datascience/jupyter/kernels/cellExecution.ts:646]: Bug fixes, ensure each stream goes into a separate output.
Concatenate only if the last output is the same as the current output stream.

File: src/client/datascience/notebook/helpers/helpers.ts [hunk 3]
```diff
@@ -364,8 +364,8 @@ export async function clearCellForExecution(editor: NotebookEditor, cell: Notebo
         const metadata = cell.metadata.with({
             statusMessage: undefined,
             executionOrder: null,
-            lastRunDuration: undefined,
-            runStartTime: undefined
+            lastRunDuration: null,
```
Comment (DonJayamanne) [inline src/client/datascience/notebook/helpers/helpers.ts:367]: VSC API changes

File: src/test/datascience/notebook/executionService.vscode.test.ts [hunk 4]
```diff
@@ -639,28 +643,48 @@ suite('DataScience - VSCode Notebook - (Execution) (slow)', function () {
...
+            {
+                metadata: {
+                    outputType: 'stream',
+                    streamName: 'stderr'
+                },
+                text: 'c'
+            }
+        ]
+        for (let index = 0; index < 4; index++) {
+            const expected = expectedOutput[index];
```
Comment (DonJayamanne) [inline src/test/datascience/notebook/executionService.vscode.test.ts:683]: Changed the test to line up with Jupyter

File: src/client/datascience/jupyter/kernels/cellExecution.ts (hunk 2 above)
Comment (DonJayamanne) [reply src/client/datascience/jupyter/kernels/cellExecution.ts:646]: Ie. do what happens in Jupyter

File: src/test/datascience/notebook/introStartPage.vscode.test.ts [hunk 5]
```diff
@@ -90,6 +90,7 @@ suite('DataScience - VSCode Notebook - Native Notebook Experiment', function ()
         await sleep(1_000);
 
         assert.isUndefined(notebookEditorProvider.activeEditor);
+        notebookEditorProvider.editors.map(item => assert.fail(item.file.fsPath, undefined, 'There should be no document open'));
```
Comment (DonJayamanne) [inline src/test/datascience/notebook/introStartPage.vscode.test.ts:93]: Logging, for some reason these tests fail on CI, the `suitesetup` ensures that all files are closed, but this fails.

**Review by rchiodo — APPROVED** [review]

**Pull request merged**